frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -141,7 +136,25 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 				#frame_support::sp_std::marker::PhantomData<(#type_use_gen,)>,
 				#frame_support::Never,
 			),
-			#( #( #[doc = #fn_doc] )* #fn_name( #( #args_compact_attr #args_type ),* ), )*
+			#(
+				#( #[doc = #fn_doc] )*
+				#fn_name {
+					#( #args_compact_attr #args_name: #args_type ),*
```

Comment:
What's the story behind the change to this new syntax for call variant structs? I'm taking a wild guess here, could it be that the new verison of the `type_info` crate makes use of the field names to create the metadata for this call variant?

Reply:
Precisely. Allows us to replace the legacy `FunctionMetadata` and `FunctionArgumentMetadata`.

Reply:
Brilliant design! Saves a lot of typing that way. There is a slight issue as a result: some field names in the call functions are prepended with an underscore (`_`) since they are not used in the function body itself, and we carry over that naming scheme here to the metadata. I wonder if the right thing is to strip away the leading underscore though, what do you think?

Reply:
yes and no, it is because we only use the type_info of the call for the metadata of the call.
In the past the metadata contained the names of the arguments, but now because the call is an enum with unnamed fields then the type info of the call doesn't contain the names of the arguments (which are important for UI).

So considering this either we implement the type info manually or we use named field in the call.
I suggested for the latter in order to avoid unnecessary manual implementation of typeinfo which is less error prone.
But I agree this implies a breaking change, fortunately it should be not hard to fix and also in general named fields is better for this enum IMO.

Reply:
Yes that is annoying. We could indeed remove the underscore, should be easy enough in the proc macro, do you have a quick tip for how to do it in legacy declarative macro?

Reply:
Hmm... I thought @thiolliere  removed the declarative macros for generating the Call enum variants in a previous PR, you shouldn't need to worry about that case... unless I'm mistaken or understanding something wrong here?

Reply:
there is still `decl_module`

Reply:
Done https://github.com/paritytech/substrate/pull/8615/commits/2b91aa8a9c7aa2a0a6a9bf41b1394323ee997c89